pkg/register/component.go

```diff
@@ -53,68 +54,96 @@ type ConfigurationDBModule interface {
 	GetConfiguration(context.Context, string) (configuration.RealmConfiguration, error)
 }
 
+// RealmRegisterConfiguration struct
+type RealmRegisterConfiguration struct {
+	Realm           string
+	TokenProvider   toolbox.OidcTokenProvider
+	EndUserGroups   []string
+	EnduserClientID string
+	SsePublicURL    string
+	endUserGroupIDs []string
+}
+
+// ComponentBuilder interface
+type ComponentBuilder interface {
+	Build() Component
+	AddTargetRealm(realmConf RealmRegisterConfiguration) error
+}
+
+type componentBuilder struct {
+	component *component
+}
+
+// NewComponentBuilder returns a builder for the management component.
+func NewComponentBuilder(keycloakURL string, keycloakClient KeycloakClient, usersDBModule keycloakb.UsersDetailsDBModule,
+	configDBModule ConfigurationDBModule, eventsDBModule database.EventsDBModule, logger internal.Logger) ComponentBuilder {
+	var component = &component{
+		keycloakURL:         keycloakURL,
+		realmConfigurations: make(map[string]RealmRegisterConfiguration),
+		keycloakClient:      keycloakClient,
+		usersDBModule:       usersDBModule,
+		configDBModule:      configDBModule,
+		eventsDBModule:      eventsDBModule,
+		logger:              logger,
+	}
+	return &componentBuilder{component: component}
+}
+
+func (r *componentBuilder) Build() Component {
+	return r.component
+}
+
+func (r *componentBuilder) AddTargetRealm(realmConf RealmRegisterConfiguration) error {
+	return r.component.addTargetRealm(realmConf)
+}
+
 // Component is the register component interface.
 type Component interface {
-	RegisterUser(ctx context.Context, clientRealmName string, user apiregister.UserRepresentation) (string, error)
+	RegisterUser(ctx context.Context, socialRealmName, clientRealmName string, user apiregister.UserRepresentation) (string, error)
```

Comment:
socialRealmName is misleading as in case of register in corporate mode, it won't be socialRealmName. Maybe should we use something like targetRealmName ? targetUserCreationRealmName ?